src/vcpkg/base/files.cpp

```diff
@@ -487,6 +487,16 @@ namespace vcpkg::Files
             Checks::exit_with_message(linfo, "error writing file: %s: %s", fs::u8string(path), ec.message());
         }
     }
+    void Filesystem::write_rename_contents(const fs::path& path,
+                                           const fs::path& tmpname,
```

Comment:
Is there a reason that this takes a tmpname? Should this ever be anything but `"{path}.{pid}.tmp"`?

Reply:
I'm not confident that's always the correct naming, so I left the api generic.